zerver/worker/thumbnail.py

```diff
@@ -120,9 +127,46 @@ def ensure_thumbnails(image_attachment: ImageAttachment) -> int:
             # We have never thumbnailed this -- it most likely had
             # bad data.  Remove the ImageAttachment row, since it is
             # not valid for thumbnailing.
+            update_message_rendered_content(
+                image_attachment.realm_id, image_attachment.path_id, None
+            )
             image_attachment.delete()
+            # TODO: Clean up any dangling thumbnails we may have
+            # produced?  Seems unlikely that we'd fail on one size,
+            # but not another, but anything's possible.
             return 0
 
     image_attachment.save(update_fields=["thumbnail_metadata"])
-
+    update_message_rendered_content(
+        image_attachment.realm_id,
+        image_attachment.path_id,
+        get_default_thumbnail_url(image_attachment),
+    )
     return written_images
+
+
+def update_message_rendered_content(
+    realm_id: int, path_id: str, image_data: tuple[str, bool] | None
+) -> None:
+    for message_class in [Message, ArchivedMessage]:
+        messages_with_image = (
+            message_class.objects.filter(  # type: ignore[attr-defined]  # TODO: ?
```

Comment:
One answer for the typing would be to make this inner loop a function that gets called twice with the two classes? Not sure.

Reply:
@andersk: Any suggestions for how to make this clean?